var/spack/repos/builtin/packages/parallel-netcdf/package.py

```diff
@@ -177,28 +177,33 @@ def configure_args(self):
 
         return args
 
-    examples_src_dir = 'examples/CXX'
+    examples_src_dir = join_path('examples', 'CXX')
 
     @run_after('install')
     def cache_test_sources(self):
         """Copy the example source files after the package is installed to an
         install test subdirectory for use during `spack test run`."""
-        self.cache_extra_test_sources([self.examples_src_dir])
+        self.cache_extra_test_sources(self.examples_src_dir)
 
     def test(self):
-        test_dir = join_path(self.install_test_root, self.examples_src_dir)
+        test_dir = join_path(self.test_suite.current_test_cache_dir,
+                             self.examples_src_dir)
         # pnetcdf has many examples to serve as a suitable smoke check.
         # column_wise was chosen based on the E4S test suite. Other
         # examples should work as well.
         test_exe = 'column_wise'
-        options = ['{0}.cpp'.format(test_exe), '-o', test_exe, '-lpnetcdf']
+        options = ['{0}.cpp'.format(test_exe), '-o', test_exe, '-lpnetcdf',
+                   '-L{0}'.format(join_path(self.prefix, 'lib')),
+                   '-I{0}'.format(join_path(self.prefix, 'include'))]
```

Comment:
Not required, but these could be simplified to the following:

```suggestion
                   '-L{0}'.format(self.prefix.lib),
                   '-I{0}'.format(self.prefix.include)]
```